clog/loggers.py

```diff
@@ -295,27 +303,66 @@ def log_line(self, stream, line):
             self.metrics.monk_exception()
 
     def _log_line_no_size_limit(self, stream, line):
-        if time.time() < self.last_timeout + self.timeout_backoff_s:
+        now = time.time()
+
+        if self.buffering:
+            if now - self.buffering_start < self.buffering_time_s:
+                self._add_to_buffer(stream, line)
+                return
+            else:
+                self.buffering = False
+                self._flush_buffer()
```

Comment:
I'm worried this might create a herd effect and flood Monk with requests once the buffering_time exceeds. Could we add a random time interval to this parameter? That might be enough to prevent this behavior.